### PR TITLE
ci: switch to GitHub Actions

### PR DESCRIPTION
**Pull request by kyleam**
The goal is for the new setup to largely follow the Drone builds, but there are some notable differences:

 * The Drone pipelines used custom images built to track Metworx's Ubuntu version and bake in many dependencies.

   The GitHub Actions use GitHub-hosted runners and test with the two most recent Ubuntu LTS releases.  r-lib/actions is used to install R and dependencies at the time of the run (with the built-in caching of r-lib/actions/setup-r-dependencies) and to execute 'R CMD check'.

 * The mpn-latest builds have been dropped.  It'd be possible to mimic that behavior by pointing the setup-r CRAN value to MPN and setting a date-pinned RSPM.  However, the mpn-latest builds arguably don't provide much value, and any unlikely mpn-latest-specific incompatibilities will be caught before release by our standard release process.  So let's avoid the extra complexity unless it becomes obvious the builds are useful.

 * 'R CMD check' is ran for the R versions matching the two most recent R versions on Metworx _and_ on the latest R version.

 * For a PR, the Drone setup had two builds, one triggered by the branch push (testing on branch tip) and another by the PR (testing on merge with base).

   For the new setup, there is just one build triggered by the PR.  It's done on the merge of the PR branch and its base.

   This means that the tip of the PR branch isn't tested.  In general, that's fine because the merge is what ultimately will land.  And for the cases where testing on the tip of the branch is useful (e.g., debugging a discrepancy due to a semantic merge conflict), a 'scratch/*' branch pointing to the same spot can be pushed.

### CODE REVIEW

**Comment by kyleam** [comment]
Test s3 auth and dry-run upload look good:

 * https://github.com/metrumresearchgroup/nmrec/actions/runs/8882642175/job/24387891988?pr=38
 * https://github.com/metrumresearchgroup/nmrec/commit/d9907ad0eec07cc4440649bd36a70fd11dc6356e

I'll drop the tip commit.

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    branches:
+      - 'main'
+      - 'scratch/**'
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+'
+      - '[0-9]+.[0-9]+.[0-9]+.[0-9]+'
+  pull_request:
+
+env:
+  MPN: 2024-03-01
```
Comment (kyleam) [inline .github/workflows/main.yaml:13]: This hard coding of the latest MPN version is unfortunate.  Drone, instead, just uses the `:latest` and `:cran-latest` Docker tags, and those get updated behind the scenes.  Each package doesn't need to handle the update.

In the near future (ideally the next MPN release), I think as part of publishing the MPN site, we should update something like `https://mpn.metworx.com/latest` to provide an easy way to programmatically get the latest MPN version.  (There is a way we've used internally for one script, but it's really convoluted, and I'd rather not propagate it to the GitHub Actions config for each of our R packages.)

Comment (kyleam) [reply .github/workflows/main.yaml]: The latest MPN is now discovered by querying a new path at `https://mpn.metworx.com`.

**Comment by kyleam** [comment]
Pushed update to pull shared actions from new repo at <https://github.com/metrumresearchgroup/actions>.

**Review by seth127 — APPROVED** [review]
Great work switching these over. Looks good here.

**Pull request merged**